packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -308,27 +311,46 @@ export class TdsDropdown {
   }
 
   private setDefaultOption = () => {
-    Array.from(this.host.children)
-      .filter((element) => element.tagName === 'TDS-DROPDOWN-OPTION')
-      .forEach((element: HTMLTdsDropdownOptionElement) => {
+    // Ensure this.host.children is not empty
+    if (this.host.children.length === 0) {
+      console.warn('TDS DROPDOWN: Data missing. Disregard if loading data asynchronously.');
+      return;
+    }
+
+    if (this.defaultValue) {
+      const children = Array.from(this.host.children).filter(
+        (element) => element.tagName === 'TDS-DROPDOWN-OPTION',
+      );
+      let matched = false;
+
+      children.forEach((element: HTMLTdsDropdownOptionElement) => {
         if (this.multiselect) {
           this.defaultValue.split(',').forEach((value) => {
             if (value === element.value) {
               element.setSelected(true);
               this.value = this.value ? [...this.value, element.value] : [element.value];
+              matched = true;
             }
           });
         } else {
           if (this.defaultValue === element.value) {
             element.setSelected(true);
             this.value = [element.value];
+            matched = true;
           } else {
             element.setSelected(false);
           }
         }
         this.setValueAttribute();
         return element;
       });
+
+      if (!matched) {
+        console.warn(
+          `TDS DROPDOWN: No matching option found for defaultValue "${this.defaultValue}"`,
```

Comment:
Really nice implementation, awesome that you found a way to handle this! 👏  When setting the defaultValue to a value that is not present in the Dropdown this warning shows up twice in the console, is the setDefaultOption function running twice some how?

Reply:
Worth checking. It might be that we would need to improve it like this (this.defaultValue !== '') { } as it seems it runs the !matched line anyway. Thanks for reporting, I will check it ;)